File: Cell.cpp
#include <iostream>
#include <SFML/Graphics.hpp>

class Cell
{
	public:
		bool state;
		float x_pos, y_pos;
		sf::RectangleShape* rectangle;

		Cell() {}

		Cell(float x_pos, float y_pos)
		{
			sf::RectangleShape *rectangle = new sf::RectangleShape();
			rectangle -> setPosition(x_pos, y_pos);
			rectangle -> setOutlineColor(sf::Color::Black);

			this -> state     = false;
			this -> x_pos     = x_pos;
			this -> y_pos     = y_pos;
			this -> rectangle = rectangle;
		}
		
		Cell(bool state, float x_pos, float y_pos, sf::RectangleShape* rectangle)
		{
			this -> state     = state;
			this -> x_pos     = x_pos;
			this -> y_pos     = y_pos;
			this -> rectangle = rectangle;
		}

		void setState(bool state)
		{
			this -> state = state;
		}

		void awake()
		{
			this -> setState(true);
			this -> rectangle -> setOutlineThickness(10000);
			this -> rectangle -> setOutlineColor(sf::Color::White);
		}

		void sleep()
		{
			this -> setState(false);
			this -> rectangle -> setOutlineThickness(0);
			this -> rectangle -> setOutlineColor(sf::Color::Black);
		}
};

File: Cell.java
class Cell
{
	private LifeState state;
	private Position prosition;
	
	
	public Cell() {}
	
	public Cell(Position prosition)
	{
		this.state     = LifeState.getRandomTerminalStates();
		this.prosition = prosition;
	}
	
	public Cell(int xPos, int yPos)
	{
		this.state     = LifeState.getRandomTerminalStates();
		this.prosition = new Position(xPos, yPos);
	}
	
	
	public LifeState getState()
	{
		return state;
	}

	public void setState(LifeState state)
	{
		this.state = state;
	}

	public Position getPosition()
	{
		return prosition;
	}
	
	public int getxPos()
	{
		return getPosition().getxPos();
	}
	
	public int getyPos()
	{
		return getPosition().getyPos();
	}

	public void setPosition(int xPos, int yPos)
	{
		this.prosition.setxPos(xPos);
		this.prosition.setyPos(yPos);
	}
	
	public boolean isAwakening()
	{
		return (this.getState() == LifeState.AWAKENING);
	}

	public boolean isAwake()
	{
		return (this.getState() == LifeState.AWAKE);
	}
	
	public boolean isSleepy()
	{
		return (this.getState() == LifeState.SLEEPY);
	}
	
	public boolean isAsleep()
	{
		return (this.getState() == LifeState.ASLEEP);
	}
}